openstudiocore/src/model/BuildingUnit.cpp

```diff
@@ -0,0 +1,430 @@
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#include <string>
+#include <memory>
+
+#include "BuildingUnit.hpp"
+#include "BuildingUnit_Impl.hpp"
+
+#include "Model.hpp"
+#include "Model_Impl.hpp"
+#include "Space.hpp"
+#include "Space_Impl.hpp"
+#include "RenderingColor.hpp"
+#include "RenderingColor_Impl.hpp"
+#include "ModelExtensibleGroup.hpp"
+
+#include <utilities/idd/OS_BuildingUnit_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/IddFactory.hxx>
+
+namespace openstudio {
+namespace model {
+namespace detail {
+
+  BuildingUnit_Impl::BuildingUnit_Impl(const IdfObject &idfObject, Model_Impl *model, bool keepHandle)
+    : ModelObject_Impl(idfObject, model, keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == BuildingUnit::iddObjectType());
+  }
+
+  BuildingUnit_Impl::BuildingUnit_Impl(const openstudio::detail::WorkspaceObject_Impl &other,
+                                       Model_Impl *model,
+                                       bool keepHandle)
+    : ModelObject_Impl(other, model, keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == BuildingUnit::iddObjectType());
+  }
+
+  BuildingUnit_Impl::BuildingUnit_Impl(const BuildingUnit_Impl &other,
+                                       Model_Impl *model,
+                                       bool keepHandle)
+    : ModelObject_Impl(other, model, keepHandle)
+  {}
+
+  const std::vector<std::string>& BuildingUnit_Impl::outputVariableNames() const
+  {
+    static std::vector<std::string> result;
+    if (result.empty()){
+    }
+    return result;
+  }
+
+  IddObjectType BuildingUnit_Impl::iddObjectType() const {
+    return BuildingUnit::iddObjectType();
+  }
+
+  boost::optional<RenderingColor> BuildingUnit_Impl::renderingColor() const
+  {
+    return getObject<ModelObject>().getModelObjectTarget<RenderingColor>(OS_BuildingUnitFields::RenderingColor);
+  }
+
+  bool BuildingUnit_Impl::setRenderingColor(const RenderingColor& renderingColor)
+  {
+    return setPointer(OS_BuildingUnitFields::RenderingColor, renderingColor.handle());
```

Comment:
i guess we will also need to extend the app, we usually do that after adding the model object